### DISCUSSION TITLE
kcadm.sh get realms/XXX does not contain "Required Actions" section parameters

### DISCUSSION THREAD

**Discussion by bazserosz31** [keycloak/keycloak#31381] [Archive]
Hi, 
We're using keycloak 21.1.2 and if I ran the following Admin Console command (kcadm.sh get realms/xxx> xxx.json), the xxx.json output file does not contain Required Actions section parameters, such as Term&Conditions etc..
![image](https://github.com/user-attachments/assets/77636f14-f3d1-4165-8cb1-b543fd2bf6ff)

Is it something which I cannot manage anymore with Admin Console? 
Thanks,
Balazs

**Comment by cturkalj** [comment]
I had to look at the browser requests sent by the Keycloak Administration UI and figure it out, because the endpoints aren't described  anywhere. Here are my experiences with Keycloak 24.0.5.

This command gets you all the (registered) required actions:
`kcadm.sh get authentication/required-actions -r myrealm`

A specific (registered) required action:
`kcadm.sh get authentication/required-actions/TERMS_AND_CONDITIONS -r myrealm`
```
{
  "alias" : "TERMS_AND_CONDITIONS",
  "name" : "Terms and Conditions",
  "providerId" : "TERMS_AND_CONDITIONS",
  "enabled" : false,
  "defaultAction" : false,
  "priority" : 20,
  "config" : { }
}

```
You can enable a registered required action like Terms&Conditions with
`kcadm.sh update "authentication/required-actions/TERMS_AND_CONDITIONS" -s enabled=true -r myrelam `

Be aware, if a required action isn't registered yet, it won't be listed under `authentication/required-actions`, you have to look at `authentication/unregistered-required-actions`. 

To enable/register it, you have to call `create authentication/register-required-action` with the body from `authentication/unregistered-required-actions`. 
 
Unfortunately `-q  providerId=myaction` doesn't not seem to work for `get authentication/unregistered-required-actions` to get the body of a specific required action.

**Reply by bazserosz31** [reply]
Awesome, many thanks! Seems like update works if I operate as authentication/required-actions. I'll manage from here.
Thanks again!
Balazs
